pandas/core/tools/datetimes.py

```diff
@@ -434,9 +449,11 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
                     yearfirst=yearfirst,
                     require_iso8601=require_iso8601
                 )
-
             if is_datetime64_dtype(result) and box:
                 result = DatetimeIndex(result, tz=tz, name=name)
+            # GH 6415
+            elif arg_is_series and utc:
```

Comment:
@jorisvandenbossche right idea but not directly related to `box`.

`_convert_listlike` can essentially be summarized in two operations:

1) If the data already has a datetime dtype, take a short cut.
2) Otherwise, make sure the data is of object dtype and try to parse.

In step 2, a Series is passed into [`_ensure_object`](https://github.com/pandas-dev/pandas/blob/648ae4f03622d8eafe1ca3b833bd6a99f56bece4/pandas/_libs/algos_common_helper.pxi.in#L549) which turns into a numpy array with object dtype. Therefore, the kludge is due to attempting to have `_convert_listlike` "remember" that a Series was passed.

But sure, I can try to move this to the `_maybe_convert_to_utc` function